### ISSUE TITLE
[gatsby-plugin-google-analytics] Pass event category, label, and action to the <OutboundLink> component

### ISSUE THREAD

**Issue by KristinaKoeva** [gatsbyjs/gatsby#20337]
## Summary

Be able to pass event category, label, and action as properties to the `<OutboundLink/>` component provided by `gatsby-plugin-google-analytics`

### Basic example

```
<OutboundLink href="mailto:careers@site.com"
    eventCategory="careers"
    eventAction="apply"
    eventLabel="office_assistance">
  <Button>Apply now</Button>
</OutboundLink>
```

### Motivation

It would be great if we can have some control over the category, action, and label of each event so that we can make Event flows based on some of those event properties.  
I would be happy to contribute if you guys think that this addition would make sense.

**Comment by freiksenet** [comment]
Thank you for the issue!

That would be a great feature indeed! Would you like to make a PR to fix that? You would need to change [OutboundLink](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-google-analytics/src/index.js#L16) component to take category/action/label from the props.

**Comment by ipiv** [comment]
There is [TrackCustomEvent Function](https://www.gatsbyjs.org/packages/gatsby-plugin-google-analytics/#trackcustomevent-function) provided by ```gatsby-plugin-google-analytics``` which allows to specify custom options, e.g **category/action/label**. 

Is this the functionality you're after?

```jsx
import React
import { trackCustomEvent } from 'gatsby-plugin-google-analytics'

export default () => {
  <div>
    <button
      onClick={e => {
        // To stop the page reloading
        e.preventDefault()
        // Lets track that custom click
        trackCustomEvent({
          // string - required - The object that was interacted with (e.g.video)
          category: "Special Button",
          // string - required - Type of interaction (e.g. 'play')
          action: "Click",
          // string - optional - Useful for categorizing events (e.g. 'Spring Campaign')
          label: "Gatsby Plugin Example Campaign",
          // number - optional - Numeric value associated with the event. (e.g. A product ID)
          value: 43
        })
        //... Other logic here
      }}
    >
      Tap that!
    </button>
  </div>
}

**Comment by KristinaKoeva** [comment]
Hey @ipiv,

At first, I thought about using this function and making a common button component in my source code that would be used when you want to track an event. But then if I do that I would be kind of copying the functionality of the `<OutboutLink>` except the ability to pass the aforementioned properties. So instead of making something custom, I decided to extend the `<OutboutLink>`. Does that make sense to you?

**Comment by KristinaKoeva** [comment]
Fixed by #20570
